sdk/database/helper/connutil/sql.go

```diff
@@ -45,7 +45,8 @@ type SQLConnectionProducer struct {
 	MaxIdleConnections       int         `json:"max_idle_connections" mapstructure:"max_idle_connections" structs:"max_idle_connections"`
 	MaxConnectionLifetimeRaw interface{} `json:"max_connection_lifetime" mapstructure:"max_connection_lifetime" structs:"max_connection_lifetime"`
 	DisableEscaping          bool        `json:"disable_escaping" mapstructure:"disable_escaping" structs:"disable_escaping"`
-	usePrivateIP             bool        `json:"use_private_ip" mapstructure:"use_private_ip" structs:"use_private_ip"`
+	UsePrivateIP             bool        `json:"use_private_ip" mapstructure:"use_private_ip" structs:"use_private_ip"`
+	UsePSC                   bool        `json:"use_psc" mapstructure:"use_psc" structs:"use_psc"`
```

Comment:
These will be returned in the READ response for all database plugins that embed this `SQLConnectionProducer` struct. That is not ideal, since they are only relevant to a small subset of database plugins. It might be better to implement this per-plugin similar to mysql. Maybe there is some common code that could be reused by each plugin?

Reply:
But this is already the case for 
```
	AuthType           string `json:"auth_type" mapstructure:"auth_type" structs:"auth_type"`
	ServiceAccountJSON string `json:"service_account_json" mapstructure:"service_account_json" structs:"service_account_json"`
```

Which is only supported for Postgresql, but still field is available for hana, mssql, redshift

Reply:
Oh actually I was able to remove from the read path, when auth_type is not gcp_iam

Test (did the same test with mysql as well)
Postgresql
Write and read using auth_type`gcp_iam`
```
vault write database/config/my-postgresql-database-psc \ 
         plugin_name="postgresql-database-plugin" \
        allowed_roles="my-role" \
connection_url="host=sej-tools-hashicorp:asia-northeast1:psc-enabled-main-instance user=cloud-sql-postgres-sa@sej-tools-hashicorp.iam dbname=postgres sslmode=disable" \
        auth_type="gcp_iam" 
        use_psc=true
Success! Data written to: database/config/my-postgresql-database-psc

vault read database/config/my-postgresql-database-psc
Key                                   Value
---                                   -----
allowed_roles                         [my-role]
connection_details                    map[auth_type:gcp_iam connection_url:host=sej-tools-hashicorp:asia-northeast1:psc-enabled-main-instance user=cloud-sql-postgres-sa@sej-tools-hashicorp.iam dbname=postgres sslmode=disable use_psc:true]
password_policy                       n/a
plugin_name                           postgresql-database-plugin
plugin_version                        n/a
root_credentials_rotate_statements    []

```

Write and read using normal plugin and normal auth
```
vault write database/config/my-postgresql-database-psc-ip      \
         plugin_name=postgresql-database-plugin \
         connection_url="postgresql://{{username}}:{{password}}@127.0.0.1/postgres?sslmode=disable"   \
         allowed_roles=readonly \
         username="admin"  \
         password="XXXXXx"
Success! Data written to: database/config/my-postgresql-database-psc-ip


vault read database/config/my-postgresql-database-psc-ip
Key                                   Value
---                                   -----
allowed_roles                         [readonly]
connection_details                    map[connection_url:postgresql://{{username}}:{{password}}@127.0.0.1/postgres?sslmode=disable username:admin]
password_policy                       n/a
plugin_name                           postgresql-database-plugin
plugin_version                        n/a
root_credentials_rotate_statements    []
```

Reply:
I was wrong about this. See my comment here https://github.com/hashicorp/vault/pull/27889/files#r1720329947